src/sentry/runner/commands/cleanup.py

```diff
@@ -398,6 +290,214 @@ def is_filtered(model: type[Model]) -> bool:
+        debug_output(">> Skipping ExportedData files")
+    else:
+        export_data_queryset = ExportedData.objects.filter(date_expired__lt=(timezone.now()))
+        for item in export_data_queryset:
+            item.delete_file()
+
+
+def models_which_use_deletions_code_path() -> list[tuple[type[Model], str, str]]:
+    from sentry.models.artifactbundle import ArtifactBundle
+    from sentry.models.eventattachment import EventAttachment
+    from sentry.models.grouprulestatus import GroupRuleStatus
+    from sentry.models.rulefirehistory import RuleFireHistory
+    from sentry.monitors.models import MonitorCheckIn
+    from sentry.replays.models import ReplayRecordingSegment
+
+    # Deletions that use the `deletions` code path (which handles their child relations)
+    # (model, datetime_field, order_by)
+    return [
+        (EventAttachment, "date_added", "date_added"),
+        (ReplayRecordingSegment, "date_added", "date_added"),
+        (ArtifactBundle, "date_added", "date_added"),
+        (MonitorCheckIn, "date_added", "date_added"),
+        (GroupRuleStatus, "date_added", "date_added"),
+        (RuleFireHistory, "date_added", "date_added"),
+    ]
+
+
+def remove_cross_project_models(
+    deletes: list[tuple[type[Model], str, str]]
+) -> list[tuple[type[Model], str, str]]:
+    from sentry.models.artifactbundle import ArtifactBundle
+
+    # These models span across projects, so let's skip them
+    deletes.remove((ArtifactBundle, "date_added", "date_added"))
+    return deletes
+
+
+def get_project_id_or_fail(project: str) -> int:
+    click.echo("Bulk NodeStore deletion not available for project selection", err=True)
+    project_id = get_project(project)
+    if project_id is None:
+        click.echo("Error: Project not found", err=True)
+        raise click.Abort()
+    return project_id
+
+
+def remove_old_nodestore_values(days: int) -> None:
+    from sentry import nodestore
+
+    debug_output("Removing old NodeStore values")
+
+    cutoff = timezone.now() - timedelta(days=days)
+    try:
+        nodestore.backend.cleanup(cutoff)
+    except NotImplementedError:
+        click.echo("NodeStore backend does not support cleanup operation", err=True)
+
+
+def generate_bulk_query_deletes() -> list[tuple[type[Model], str, str | None]]:
+    from django.apps import apps
+
+    from sentry.models.groupemailthread import GroupEmailThread
+    from sentry.models.userreport import UserReport
+
+    # Deletions that use `BulkDeleteQuery` (and don't need to worry about child relations)
+    # (model, datetime_field, order_by)
+    additional_bulk_query_deletes = []
+    for entry in settings.ADDITIONAL_BULK_QUERY_DELETES:
+        app_name, model_name = entry[0].split(".")
+        model_tp = apps.get_model(app_name, model_name)
+        additional_bulk_query_deletes.append((model_tp, entry[1], entry[2]))
+
+    BULK_QUERY_DELETES = [
+        (UserReport, "date_added", None),
+        (GroupEmailThread, "date", None),
```

Comment:
I've made sure to port the changes from this PR:
https://github.com/getsentry/sentry/pull/77289/files#diff-af1cde82f065157a87af5b9c019c93a9e93aeb22661494449dbca1628e3cf306R212